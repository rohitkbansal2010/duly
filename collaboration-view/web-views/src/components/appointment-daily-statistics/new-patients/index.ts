export { NewPatients } from './new-patients';
