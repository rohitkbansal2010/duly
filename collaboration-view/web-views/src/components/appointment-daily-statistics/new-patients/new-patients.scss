@import "~@styles/colors.scss";
@import "~@styles/media-mixins.scss";
@import "~@styles/spacing-constants.scss";
@import "~@styles/main.scss";
@import "~@styles/typography.scss";
@import "~@styles/devices.scss";

.newPatientsRow {
  font-size: 1.5rem;
  line-height: 1.8rem;
  white-space: nowrap;
  @include nexa-heavy-highlight;

  &Container {
    position: relative;
    padding-left: 3.2rem !important;
  }

  & img {
    position: absolute;
    bottom: 0;
    left: 0.5rem;
    width: 2.6rem;
    padding: 0 0.3rem;
  }

  &Count {
    color: $dupage-dark-blue;
  }

  &Label {
    margin-left: 0.4rem;
    color: $medium-gray;
  }
}
