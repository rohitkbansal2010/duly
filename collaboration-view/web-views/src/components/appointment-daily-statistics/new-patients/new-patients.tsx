import { Row } from 'react-bootstrap';

import { newPatientsMagenta } from '@icons';
import { DailyStatistics } from '@types';

import styles from './new-patients.scss';

type NewPatientsTypeProps = Pick<DailyStatistics, 'newPatients'>;

export const NewPatients = ({ newPatients }: NewPatientsTypeProps) => {
  if (!newPatients) return null;

  const newPatientsLabel = newPatients > 1 ? 'New Patients' : 'New Patient';

  return (
    <Row className={styles.newPatientsRow}>
      <span className={styles.newPatientsRowContainer}>
        <img src={newPatientsMagenta} alt="new patients icon" />
        <span className={styles.newPatientsRowCount}>{newPatients}</span>
        <span className={styles.newPatientsRowLabel}>{newPatientsLabel}</span>
      </span>
    </Row>
  );
};
