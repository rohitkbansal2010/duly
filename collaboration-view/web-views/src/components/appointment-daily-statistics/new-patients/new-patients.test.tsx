import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { NewPatients } from './new-patients';

describe('NewPatients', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<NewPatients newPatients={5}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
