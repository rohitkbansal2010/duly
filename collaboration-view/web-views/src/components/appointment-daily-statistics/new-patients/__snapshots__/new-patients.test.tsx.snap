// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPatients should match snapshot 1`] = `
<Row
  className="newPatientsRow"
>
  <span
    className="newPatientsRowContainer"
  >
    <img
      alt="new patients icon"
      src="new-patients-magenta.svg"
    />
    <span
      className="newPatientsRowCount"
    >
      5
    </span>
    <span
      className="newPatientsRowLabel"
    >
      New Patients
    </span>
  </span>
</Row>
`;
